launchable/commands/subset.py

```diff
@@ -159,7 +159,14 @@ def default_path_builder(file_name):
 
             for t in glob.iglob(join(base, pattern), recursive=True):
                 if path_builder:
-                    path = path_builder(t[len(base)+1:])
+                    if glob.has_magic(base):
+                        for b in glob.glob(base):
```

Comment:
This fores `glob.glob(base)` to be evaluated repeatedly which is costly and unnecessary.

How about we evaluate `glob.igob(base)` then for each path it returns we evaluate `glob.iglob(x,pattern)` which minimizes directory traversal?

Reply:
Yes, indeed.
I'll fix it with your approach